modDbImpl/src/org/aion/db/generic/DatabaseWithCache.java

```diff
@@ -425,69 +425,81 @@ public boolean isEmpty() {
 
     @Override
     public void put(byte[] k, byte[] v) {
-        AbstractDB.check(k);
-
-        check();
-
-        ByteArrayWrapper key = ByteArrayWrapper.wrap(k);
-
-        this.loadingCache.put(key, Optional.ofNullable(v));
-        // keeping track of dirty data
-        this.dirtyEntries.put(key, v);
+        putToBatch(k, v);
 
+        // also checks if flush is needed
         if (enableAutoCommit) {
             flushInternal();
         }
     }
 
     @Override
     public void delete(byte[] k) {
-        // put also handles synchronization
-        put(k, null);
+        deleteInBatch(k);
+
+        // also checks if flush is needed
+        if (enableAutoCommit) {
+            flushInternal();
+        }
     }
 
     @Override
-    public void putBatch(Map<byte[], byte[]> inputMap) {
-        AbstractDB.check(inputMap.keySet());
-
+    public void putToBatch(byte[] k, byte[] v) {
+        AbstractDB.check(k);
+        AbstractDB.check(v);
         check();
 
-        for (Map.Entry<byte[], byte[]> entry : inputMap.entrySet()) {
-            ByteArrayWrapper key = ByteArrayWrapper.wrap(entry.getKey());
-            byte[] value = entry.getValue();
+        ByteArrayWrapper key = ByteArrayWrapper.wrap(k);
```

Comment:
any benefit for using ByteArrayWrapper in this Cache class, can we use byte[] directly to avoid twice conversion?

Reply:
@AlexandraRoatis would have better insight into this. I think it's to take advantage of the `equals` mostly?